os/mm/mm_heap/mm_manage_allocfail.c

```diff
@@ -130,9 +136,48 @@ void mm_manage_alloc_fail(struct mm_heap_s *heap, int startidx, int endidx, size
 	mfdbg(" - total free size   : %d\n", info.fordblks);
 
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	int cur_app_idx;
+	if (heap_type != KERNEL_HEAP) {
+		for (int i = 1; i <= CONFIG_NUM_APPS; i++) {
+			if (heap == BIN_BINARY_HEAP_PTR(i)) {
+				cur_app_idx = i;
+			}
+		}
```

Comment:
You can get the app idx from the app_id variable in current tcb. 
Please change above code.